dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedScoreTracker.java

```diff
@@ -151,10 +151,10 @@ public void undoStartRequest() {
         public void onSuccess(Response response) {
             inflight.decrementAndGet();
 
-            if (Responses.isQosStatus(response) || Responses.isServerError(response)) {
+            if (isGlobalQosStatus(response) || Responses.isServerErrorRange(response)) {
```

Comment:
this change should have a pretty big impact on node-selection decisions for our internal ~ski product~ atlasdb proxy :+1: